File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
        
    answer = []
    t_y, t_m, t_d = map(int, today.split("."))
    idx = 1
    for priv in privacies:
        date, typ = map(str, priv.split())
        y, m, d = map(int, date.split("."))
        for i in terms:
            a, b = map(str, i.split())
            if typ == a:
                m += int(b)
        y += m//12
        m = m%12

        res = ((t_y-y)*12 + t_m-m)*30 + (t_d-d)
        if res >= 0:
            answer.append(idx)
        idx += 1
        
    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.java
import java.util.*;



class Solution {    
    public List<Integer> solution(String today, String[] terms, String[] privacies) {
        
        List<Integer> answer = new ArrayList<>();

        String[] tds = today.split("\\.");
        int tY = Integer.parseInt(tds[0]);
        int tM = Integer.parseInt(tds[1]);
        int tD = Integer.parseInt(tds[2]);
        
        int idx = 1;
        
        for (String priv : privacies) {

            // privacies, 시작 날짜
            String[] item = priv.split(" ");
            String date = item[0];
            String type = item[1];
            
            String[] startDate = date.split("\\.");
            int year = Integer.parseInt(startDate[0]);
            int month = Integer.parseInt(startDate[1]);
            int day = Integer.parseInt(startDate[2]);

            for (String ty : terms) {
                // terms, 유효기간
                String[] tmp = ty.split(" ");
                String _type = tmp[0];
                int m = Integer.parseInt(tmp[1]);
                
                if (type.equals(_type)) {
                    month += m;
                    break;
                }
            }
            
            year += month/12;
            month = month%12;
            if (month == 0) {
                year -= 1;
                month = 12;
            }
            
            int total = (tY-year)*12*28 + (tM-month)*28 + (tD-day);
            if (total >= 0) {
                answer.add(idx);
            }   
            idx++;
        }

        return answer;
    }
}